espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/model/SourceDocument.java

```diff
@@ -127,6 +134,27 @@ private void buildXML(Document doc, Element parentElement, View view) {
             parentElement.appendChild(element);
         }
 
+        // If it's an AdapterView, get the adapters as a String
+        if (view instanceof AdapterView) {
+            AdapterView adapterView = (AdapterView) view;
+            Adapter adapter = adapterView.getAdapter();
+            StringBuilder adapterData = new StringBuilder();
+            for (int i = 0; i < adapter.getCount(); i++) {
+                Object adapterItem = adapter.getItem(i);
+                adapterData.append(adapterItem);
+                if (i < adapter.getCount() - 1) {
+                    adapterData.append(",");
+                }
+
+                // Get the type of the adapter item
+                if (i == 0) {
+                    setAttribute(element, ViewAttributesEnum.ADAPTER_TYPE, adapterItem.getClass().getSimpleName());
+                }
+                setAttribute(element, ViewAttributesEnum.ADAPTERS, adapterData);
```

Comment:
adapterData.toString()

Reply:
It seems to have serialized it correctly without using `toString()`

Reply:
yeah, I know, but explicit is always better than implicit